code/nebula/neb.cpp

```diff
@@ -191,62 +201,109 @@ void neb2_init()
 
 		// poofs
 		while (required_string_one_of(3, "#end", "+Poof:", "$Name:")) {
+			bool create_if_not_found = true;
+			poof_info pooft;
+			poof_info* poofp;
+			bool poof_new = true;
 
-			if (Poof_info.size() < MAX_NEB2_POOFS) {
-				poof_info new_poof;
+			if (optional_string("+Poof:")) { // retail style
+				stuff_string(name, F_NAME, MAX_FILENAME_LEN);
+				strcpy_s(pooft.bitmap_filename, name);
 
-				if (optional_string("+Poof:")) { // retail style
-					stuff_string(name, F_NAME, MAX_FILENAME_LEN);
-					strcpy_s(new_poof.bitmap_filename, name);
+				strcpy_s(pooft.name, name);
 
-					strcpy_s(new_poof.name, name);
+				generic_anim_init(&pooft.bitmap, name);
 
-					generic_anim_init(&new_poof.bitmap, name);
+				if (Poof_info.size() < MAX_NEB2_POOFS) {
+					Poof_info.push_back(pooft);
+				} else {
+					mprintf(("Warning: more than 32 poofs are defined! Skipping poof %s\n", pooft.name));
```

Comment:
This (and the similar check below) should probably be an actual warning

Reply:
Done